internal/internal_test.go

```diff
@@ -1272,12 +1274,145 @@ func TestGetParsedBody(t *testing.T) {
 		}
 	  }`
 
-	_, _, err := getParsedBody(createCheckRequest(requestContentTypeJSONInvalid))
+	path := []interface{}{}
+	protoDescriptorPath := ""
+
+	_, _, err := getParsedBody(createCheckRequest(requestContentTypeJSONInvalid), path, protoDescriptorPath)
 	if err == nil {
 		t.Fatal("Expected error but got nil")
 	}
 }
 
+func TestGetParsedBodygRPC(t *testing.T) {
+
+	requestValidExample := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/grpc"
+			  },
+			  "method": "POST",
+			  "path": "/Example.Test.GRPC.ProtoServiceIExampleApplication/RegisterExample",
+			  "protocol": "HTTP/2",
+			  "raw_body": "AAAAADYKFgoHCgVFUlJPUhILCglTZWNOdW1iZXISHAoMCgpCb2R5IHZhbHVlEgwKCk5hbWUgVmFsdWU=",
+			}
+		  },
+		  "parsed_path": [
+			"Example.Test.GRPC.ProtoServiceIExampleApplication",
+			"RegisterExample"
+		  ]
+		}
+	  }`
+	requestValidBook := `{
+	   	"attributes": {
+	   	  "request": {
+	   		"http": {
+	   		  "headers": {
+	   			"content-type": "application/grpc"
+	   		  },
+	   		  "method": "POST",
+         		  "path": "/com.book.BookService/GetBooksViaAuthor",
+                 "protocol": "HTTP/2",
+                 "raw_body": "AAAAAAYKBEpvaG4="
+	   		}
+	   	  }
+	   	}
+	    }`
+
+	requestInvalidRawBodyExample := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/grpc"
+			  },
+			  "method": "POST",
+			  "path": "/Example.Test.GRPC.ProtoServiceIExampleApplication/RegisterExample",
+			  "protocol": "HTTP/2"
+			}
+		  },
+		  "parsed_path": [
+			"Example.Test.GRPC.ProtoServiceIExampleApplication",
+			"RegisterExample"
+		  ]
+		}
+	  }`
+
+	requestInvalidParsedPathExample := `{
+		"attributes": {
+		  "request": {
+			"http": {
+			  "headers": {
+				"content-type": "application/grpc"
+			  },
+			  "method": "POST",
+			  "protocol": "HTTP/2",
+			  "raw_body": "AAAAAC0KDQoHCgVFUlJPUhICCAESHAoMCgpCb2R5IHZhbHVlEgwKCk5hbWUgVmFsdWU="
+			}
+		  }
+		}
+	  }`
+
+	expectedObject1 := map[string]interface{}{}
+	expectedObject1["Body"] = "Body value"
+	expectedObject1["Name"] = "Name Value"
+
+	expectedObject2 := map[string]interface{}{}
+	expectedObject2["SeverityNumber"] = "SecNumber"
+	expectedObject2["SeverityText"] = "ERROR"
+
+	expectedObject := map[string]interface{}{}
+	expectedObject["Data"] = expectedObject1
+	expectedObject["Metadata"] = expectedObject2
+
+	protoDescriptorExamplePath := `../test/files/example/example.pb`
```

Comment:
The test failure is because the tests are run on linux: the filesystem is case-sensitive. The committed `example.pb` is in `test/files/Example/example.pb`, and referenced in the test code as `test/files/example/example.pb`. That works fine on case-insensitive FS (OSX, maybe Windows?) but fails in travis.

Reply:
Can you make the committed path lowercase, please? It's more common, I think.

Reply:
ok changed it. The tests are successful now